src/nbl/video/COpenGL_LogicalDevice.h

```diff
@@ -225,7 +252,65 @@ class COpenGL_LogicalDevice : public IOpenGL_LogicalDevice
     {
         return core::make_smart_refctd_ptr<IDescriptorPool>(core::smart_refctd_ptr<IOpenGL_LogicalDevice>(this),maxSets);
     }
+    
+    SMemoryOffset allocate(const SAllocateInfo& info) override
+    {
+        SMemoryOffset ret =  {nullptr, IDeviceMemoryAllocator::InvalidMemoryOffset};
+        if(info.dedication)
+        {
+            IOpenGLMemoryAllocation* glAllocation = nullptr;
+            if(info.dedication->getObjectType() == IDriverMemoryBacked::EOT_BUFFER)
+            {
+                COpenGLBuffer* buffer = static_cast<COpenGLBuffer*>(info.dedication);
+                glAllocation = static_cast<IOpenGLMemoryAllocation*>(buffer);
+            }
+            else if(info.dedication->getObjectType() == IDriverMemoryBacked::EOT_IMAGE)
+            {
+                COpenGLImage* image = static_cast<COpenGLImage*>(info.dedication);
+                glAllocation = static_cast<IOpenGLMemoryAllocation*>(image);
+            }
+
+            if(info.memoryTypeIndex < m_physicalDevice->getMemoryProperties().memoryTypeCount)
+            {
+                SRequestAllocate reqParams;
+                reqParams.dedicationAsAllocation = glAllocation;
+                reqParams.memoryAllocateFlags = core::bitflag<IDriverMemoryAllocation::E_MEMORY_ALLOCATE_FLAGS>(info.flags);
+                reqParams.memoryPropertyFlags = m_physicalDevice->getMemoryProperties().memoryTypes[info.memoryTypeIndex].propertyFlags;
+
+                auto& req = m_threadHandler.request(std::move(reqParams),&ret);
+                m_masterContextCallsInvoked++;
+                m_threadHandler.template waitForRequestCompletion<SRequestAllocate>(req);
+                assert(ret.memory && ret.offset != IDeviceMemoryAllocator::InvalidMemoryOffset);
+            }
+            else
+            {
+                assert(false);
+            }
+        }
+        else
+        {
+            assert(false);
+        }
+        return ret;
+    }
+
+    core::smart_refctd_ptr<IGPUBuffer> createBuffer(const IGPUBuffer::SCreationParams& creationParams) override
+    {
+        SRequestBufferCreate2 reqParams;
+        reqParams.mreqs.size = creationParams.declaredSize;
+        reqParams.mreqs.memoryTypeBits = 0xffffffffu;
+        reqParams.mreqs.alignmentLog2 = 0u; // TODO(Erfan) Alignment previously was 0u in getXXXMemoryRequirementsOnDedMem(). what to set here? get minXXXOffsetAlignment from physical device and deduce from usage?
+        reqParams.mreqs.prefersDedicatedAllocation = true;
+        reqParams.mreqs.requiresDedicatedAllocation = true;
```

Comment:
maybe don't send stuff you don't need XD

Reply:
Yea I think It's better to write most of these stuff when processing the requests